contents/javascript/Javascript-promises.md

```diff
@@ -0,0 +1,104 @@
+# Javascript: Promises
+
+Author: Daniel Berzin Chua
+
+In Javascript, there are occasions where a method call may not resolve instantly and as such, the code may not run in the sequence that you would expect due this asynchronous nature. It can be especially confusing for beginners learning Javascript.
+
+A pertinent example would be HTTP requests. When making a HTTP request, the information from the server is not instantly returned as there is a round trip time that involves the time taken for the request to be sent, and for the response from the server to be sent back to the client. In this time, Javascript would not necessarily wait for the response to be received, but instead execute other functions first, even if those functions depend on the result of that HTTP response.
```

Comment:
good example. a simpler example would actually be timer, i.e. `setTimeout()`. You can explain what is callback straight up here in a setTimeout example and then talk about async execution and event loop

Reply:
The example is good and easy to understand for me